web3/middleware/filter.py

```diff
@@ -277,6 +277,8 @@ def get_filter_changes(self):
 @to_list
 def block_hashes_in_range(w3, block_range):
     from_block, to_block = block_range
+    if from_block is None or to_block is None:
+        return None
```

Comment:
I think this should just be `return`.  The *value* from a `return` statement in a generator doesn't actually get returned.

Reply:
If we get `flake8-bugbear` added to the lint run for this repo it will catch things like this (by explicitely enabling [`B901`](https://github.com/PyCQA/flake8-bugbear#opinionated-warnings))